sky/packages/sky/lib/src/material/scrollbar_painter.dart

```diff
@@ -0,0 +1,103 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui' as ui;
+
+import 'package:flutter/animation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+const double _kMinScrollbarThumbLength = 18.0;
+const double _kScrollbarThumbGirth = 6.0;
+const Duration _kScrollbarThumbFadeDuration = const Duration(milliseconds: 300);
+
+class ScrollbarPainter extends ScrollableListPainter {
+
+  double get opacity => _opacity;
+  double _opacity = 0.0;
+  void set opacity (double value) {
```

Comment:
is this meant to be internal state or external state? You seem to provide an API for it to be changed by this object's owner, but then this object also tries to mutate it itself. What's the contract for this field?